src/main/scala/viper/silver/plugin/standard/termination/transformation/ExpTransformer.scala

```diff
@@ -182,6 +214,27 @@ trait ExpTransformer extends ErrorReporter {
     case Forall(_, _, exp) => Seq(exp)
   }
 
+  /**
+    * Turns `vars` into new local variable declarations with a unique name and replaces their occurrences in `exp`.
+    * The new local variables declarations and the transformed expression are returned
+    */
+  protected def substituteWithFreshVars(vars: Seq[LocalVarDecl], exp: Exp): (Seq[LocalVarDecl], Exp) = {
+    val declMapping = vars.map(decl => decl -> uniqueName(decl.name))
+    val freshDecls = declMapping.map {
+      case (oldDecl, freshName) => LocalVarDecl(freshName, oldDecl.typ)(oldDecl.pos, oldDecl.info, oldDecl.errT)
+    }
+    val transformedExp = exp.transform({
+      case v@LocalVar(name, typ) => declMapping
+        // check whether the local variable is in the map of variables that should be replaced:
+        .collectFirst { case (decl, freshName) if decl.name == name => freshName }
+        // replace it by a new local variable:
+        .map(freshName => LocalVar(freshName, typ)(v.pos, v.info, v.errT))
+        // keep the variable unchanged if no replacement should happen:
+        .getOrElse(v)
+    }, Traverse.TopDown)
+    (freshDecls, transformedExp)
+  }
```

Comment:
Silver seems to have some functionality already to perform variable substitutions (viper.silver.ast.utility.Expressions). Couldn't we use it instead?